### PR TITLE
Fix getp1 for dynamic arrays

### PR DESCRIPTION
**Pull request by szymonlopaciuk**
Supersedes #59
Requires #62 
Closes xsuite/xsuite#178

- Add a missing semicolon in `capi.Index_get_c_offset`
- Correct the calculation of the right offset (do not discard the value)
- Change the return in `capi.gen_c_pointed` so that it does not try to dereference the value of the string

### CODE REVIEW

File: tests/test_capi.py [hunk 1]
```diff
@@ -12,71 +12,6 @@
 ffi = cffi.FFI()
 
 
-def gen_classes():
```
Comment (rdemaria) [inline tests/test_capi.py:15]: why did you remove this?

Comment (szymonlopaciuk) [reply tests/test_capi.py:15]: It seems like it's not used anywhere in the tests

Comment (rdemaria) [reply tests/test_capi.py:15]: It was a reminder to add tests ;-)  Could you try to add a basic initialization of each one of the classes?

File: xobjects/capi.py [hunk 2]
```diff
@@ -202,7 +203,7 @@ def gen_fun_kernel(cls, path, action, const, extra, ret, add_nindex=False):
 def gen_c_pointed(target: Arg, conf):
     size = gen_c_size_from_arg(target, conf)
     ret = gen_c_type_from_arg(target, conf)
-    if target.pointer or is_compound(target.atype):
+    if target.pointer or is_compound(target.atype) or is_string(target.atype):
```
Comment (rdemaria) [inline xobjects/capi.py:206]: ok for me
